src/dapla_metadata/datasets/core.py

```diff
@@ -545,3 +616,27 @@ def percent_complete(self) -> int:
             self.dataset,
         ) + num_obligatory_variables_fields_completed(self.variables)
         return calculate_percentage(num_set_fields, num_all_fields)
+
+    def add_pseudo_variable(self, variable_short_name: str) -> None:
+        """Adds a new pseudo variable to the list of pseudonymized variables."""
+        if self.variables_lookup[variable_short_name] is not None:
+            self.pseudo_variables.append(
+                model.PseudoVariable(short_name=variable_short_name)
+            )
+
+    def get_pseudo_variable(
+        self, variable_short_name: str
+    ) -> model.PseudoVariable | None:
+        """Finds a pseudo variable by shortname."""
+        if self.variables_lookup.get(variable_short_name):
+            return next(
+                (
+                    v
+                    for v in self.pseudo_variables
+                    if v.short_name == variable_short_name
+                ),
+                None,
+            )
+
+        msg = f"No pseudo variable was found with short_name={variable_short_name}"
+        raise KeyError(msg)
```

Comment:
```suggestion
        return self.pseudo_variables_lookup.get(variable_short_name, None)
```

Do we need to check that the short name is in the variables list here? It feels like it's more important to do that when we create it.